domain/domain.go

```diff
@@ -199,6 +200,20 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 	if err != nil {
 		return nil, false, 0, nil, err
 	}
+	// fetch the commit timestamp of the schema diff
+	schemaTs := int64(0)
+	tikvStore, ok := do.Store().(helper.Storage)
+	if ok {
+		helper := helper.NewHelper(tikvStore)
+		schemaTs, err = m.GetTimestampForSchemaVersion(helper, neededSchemaVersion)
```

Comment:
Rename `GetTimestampForSchemaVersion`  to `GetTimestampForSchemaVersionWithNonEmptyDiff` and coment that `GetTimestampForSchemaVersionWithNonEmptyDiff` should not fail if `GetSchemaVersionWithNonEmptyDiff` did not fail.